sdk/simulator-service/src/strategies/refinance/Refinance.ts

```diff
@@ -1,70 +1,32 @@
 import {
   FlashloanProvider,
-  Simulation,
+  ISimulation,
   SimulationSteps,
   SimulationType,
 } from '@summerfi/sdk-common/simulation'
-import { getReferencedValue, makeStrategy } from '../helpers'
-import { Simulator } from '../simulator-engine'
-import { Percentage, TokenAmount } from '@summerfi/sdk-common/common'
+import { getReferencedValue } from '../../implementation/utils'
+import { Simulator } from '../../implementation/simulator-engine'
+import { TokenAmount } from '@summerfi/sdk-common/common'
 import { newEmptyPositionFromPool } from '@summerfi/sdk-common/common/utils'
 import { RefinanceParameters } from '@summerfi/sdk-common/orders'
-import { type ISwapManager } from '@summerfi/swap-common/interfaces'
 import { isLendingPool } from '@summerfi/sdk-common/protocols'
+import { refinanceLendingToLendingStrategy } from './Strategy'
+import { RefinanceDependencies } from './Types'
 
-export const refinanceStrategy = makeStrategy([
-  {
-    step: SimulationSteps.Flashloan,
-    optional: false,
-  },
-  {
-    step: SimulationSteps.PaybackWithdraw,
-    optional: false,
-  },
-  {
-    step: SimulationSteps.Swap,
-    optional: true,
-  },
-  {
-    step: SimulationSteps.DepositBorrow,
-    optional: false,
-  },
-  {
-    step: SimulationSteps.Swap,
-    optional: true,
-  },
-  {
-    step: SimulationSteps.RepayFlashloan,
-    optional: false,
-  },
-  {
-    // In case of target debt being different then source debt we need a swap,
-    // We cannot forsee the exact amount of the swap, so we need to return excess tokens to user
-    step: SimulationSteps.ReturnFunds,
-    optional: true,
-  },
-])
-
-// TODO move those interfaces to more appropriate place
-
-export interface RefinanceDependencies {
-  swapManager: ISwapManager
-  getSummerFee: () => Percentage
-}
-
-export async function refinaceLendingToLending(
+export async function refinance(
```

Comment:
this was named refinaceLendingToLending on purpose, as it is not a generic refinance